opensipscli/modules/diagnose.py

```diff
@@ -193,10 +196,37 @@ class diagnose(Module):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.t = None
+        self.__rcv_proto = 'tcp'
+        self.__rcv_ip = '127.0.0.1'
+        self.__rcv_port = 8888
+        if cfg.exists("jsonrpc_listen_ip"):
+            self.__rcv_ip = cfg.get("jsonrpc_listen_ip")
```

Comment:
The `jsonrpc_listen_ip` and `jsonrpc_listen_port` are a good idea, however it seems there is already support within the config file for the JSON-RPC destination, see the [`url`](https://github.com/OpenSIPS/opensips-cli/blob/master/opensipscli/defaults.py#L67) param.  I think it's better if you parse `cfg.get('url')` at the top, using `urllib.parse.urlparse()` and extract the `.hostname` and `.port` members.  This way, developers & sysadmins will have 2 less settings to worry about ;)

Reply:
I looked at `url` parameter and I actually use it. And it is for OpenSIPS side json-rpc. Host:port where OpenSIPS listens and opensips-cli sends commands. But new parameters is host:port where opensips-cli listens for json-rpc events from OpenSIPS, so they are different hosts in general case. May be I need to change defaults to another port, which makes it confusing.

Reply:
@vasilevalex, sorry for misleading you.  The `jsonrpc_listen_ip` name confused me, since I thought it was on the OpenSIPS side.  So these 2 extra parameters are needed **for sure** now.  There are 2 "nice-to-have" TODOs left, as I see it:

1) instead of providing their default values under the `__init__()` method, maybe a more uniform approach would be to update the `opensipscli/defaults.py` file with these values, for the "diagnose" module.  This would make them a lot more visible, for everyone.

2) maybe rename these to something more obvious?  For example: `diagnose_listen_ip`, `diagnose_listen_port`

PS: I noticed you changed from 8888 -> 8899, so it works by default ;)

Reply:
Sorry, I have missed your last edit...